runtime/onert/core/src/util/shapeinf/Transpose.cc

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+#include <iostream>
+namespace onert
+{
+namespace shape_inference
+{
+// helper function
+ir::Shape inferTransposeShape(const ir::Shape &in_shape, const std::vector<int> &perm)
+{
+  if (static_cast<int>(perm.size()) > in_shape.rank())
+  {
+    throw std::runtime_error("inferTransposeShape failed, bad rank size: " +
+                             std::to_string(static_cast<int>(perm.size())));
+  }
+  ir::Shape out_shape(static_cast<int>(perm.size()));
+  for (int idx = 0; idx < static_cast<int>(perm.size()); idx++)
+  {
+    if (perm[idx] < 0 || perm[idx] >= static_cast<int>(perm.size()))
+    {
+      throw std::runtime_error("inferTransposeShape failed, bad perm value: " +
+                               std::to_string(perm[idx]));
+    }
+    out_shape.dim(idx) = in_shape.dim(perm[idx]);
+  }
+  return out_shape;
+}
+
+// StaticInferer at compilation time
+void StaticInferer::visit(const ir::operation::Transpose &op)
+{
+  const auto input_idx{op.getInputs().at(ir::operation::Transpose::Input::INPUT)};
+  const auto &input = _operands.at(input_idx);
+
+  // get mutable output operand
+  const auto output_idx = op.getOutputs().at(0);
+  ir::Operand &output = _operands.at(output_idx);
+  const auto perm{op.param().perm};
+  // const auto rank{op.param().rank};
+  // if input is dynamic, or perm vector is not yet filled, output also becomes dynamic
```

Comment:
Could you check `perm vector is not yet filled` is correct comment? IMHO, our assumption here is that `perm` is always filled.

Reply:
I will remove the comment line regarding `perm vector`. We do not check for it now, so the comment is out of place. Thanks.